src/verify_ce_config.py

```diff
@@ -103,35 +108,45 @@ def main():
                          "but cannot be found in JOB_ROUTER_ENTRIES: %s"
                          % ', '.join(missing_route_def))
 
-    # Find all eval_set_ attributes in the JOB_ROUTER_DEFAULTS
-    eval_set_defaults = set([x.lstrip('eval_') for x in parsed_jr_ads['JOB_ROUTER_DEFAULTS'][0].keys()
-                             if x.startswith('eval_set_')])
-
-    # Find all default_ attributes used in expressions in the JOB_ROUTER_DEFAULTS
-    default_attr = set([re.sub(r'.*(default_\w*).*', 'eval_set_\\1', str(x))
-                        for x in parsed_jr_ads['JOB_ROUTER_DEFAULTS'][0].values()
-                        if isinstance(x, classad.ExprTree) and "default_" in str(x)])
-
-    for entry in parsed_jr_ads['JOB_ROUTER_ENTRIES']:
-        # Warn users if they've set_ attributes that would be overriden by eval_set in the JOB_ROUTER_DEFAULTS
-        overriden_attr = eval_set_defaults.intersection(set(entry.keys()))
-        if overriden_attr:
-            warn("%s in JOB_ROUTER_ENTRIES will be overriden by the JOB_ROUTER_DEFAULTS."
-                 % ', '.join(overriden_attr)
-                 + " Use the 'eval_set_' prefix instead.")
-
-        # Ensure that users don't set the job environment in the Job Router
-        if is_osg and any(x.endswith('environment') for x in entry.keys()):
-            error("Do not use the Job Router to set the environment. See documentation for more details: "
-                  + "https://htcondor.github.io/htcondor-ce/v5/configuration/writing-job-routes/#setting-job-environments")
-
-        # Warn users about eval_set_ default attributes in the ENTRIES since their
-        # evaluation may occur after the eval_set_ expressions containg them in the
-        # JOB_ROUTER_DEFAULTS
-        no_effect_attr = default_attr.intersection(set([x for x in entry.keys() if x.startswith('eval_set_')]))
-        if no_effect_attr:
-            warn("%s in JOB_ROUTER_ENTRIES " % ', '.join(no_effect_attr)
-                 + "may not have any effect. Use the 'set_' prefix instead.")
+    # Also warn about routes specified in JOB_ROUTER_ROUTE_<name> transforms that don't appear in the parsed JRE.
+    if jr_transforms:
+        for transform_attr, transform_value in jr_transforms:
+            missing_route_def = set(transform_value.replace(',',' ').split()).difference(set(parse_route_names(config_val)))
+            if missing_route_def:
+                warn("The following are specified in %s "
+                        "but cannot be found in JOB_ROUTER_ENTRIES: %s"
+                        % transform_attr, ', '.join(missing_route_def))
```

Comment:
This isn't quite the right check to make (transforms are separate from entries). We should warn if there are route names showing up in `JOB_ROUTER_ROUTE_NAMES` that don't have a corresponding `JOB_ROUTER_ROUTE_<NAME>` config variable

Also, the usage of `config_val` will likely lead to confusion down the line since it's defined in the for loop and it luckily works here since the last iteration in the loop will set `config_val` the ads defined by `JOB_ROUTER_ENTRIES`